packages/view-types/link-controller/src/LinkController.js

```diff
@@ -0,0 +1,143 @@
+        newConfig.description, fileTypes, coordinationTypes, stores
+      )});
+  }, [viewConfigStoreApi]);
+
+  const onConfigChange = useCallback((configValue) => {
+    // console.log(send, socketOpen)
+    if (sync && socketOpen) {
+      try {
+        //console.log(configValue)
+        let configString = id + ';' + btoa(escape(JSON.stringify(configValue)));
+        //console.log("Send Message")
+        connection.current?.send('{"action": "sendMessage","message": "' + configString + '"}');
+        // connection.current?.send('{"action": "sendMessage","message": "mesage"}');
+      } catch (error) {
+        console.log(error)
+      }
+    }
+  }, [id, socketOpen, sync]);
+
+
+  useEffect(() => {
+    if (linkIDInit != null) {
+      setLinkID(linkIDInit)
+    }
+    if (linkID == null) {
+      fetch("https://nwe7zm1a12.execute-api.us-east-1.amazonaws.com/link", {
+        method: 'GET',
+        headers: {
+          'Accept': 'application/json',
+        }
+      }).then(response => response.json()).then(function (response) {
+        setLinkID(response.link_id);
+      }).catch(function (err) {
+        console.log('Fetch Error :-S', err);
+      });
+    }
+  }, [linkID])
+
+  useEffect(() => viewConfigStoreApi.subscribe(
+    // The function to run on each publish.
+    ({viewConfig, mostRecentConfigSource}) => {
+      //console.log(viewConfig, mostRecentConfigSource)
+      if (onConfigChange && viewConfig && mostRecentConfigSource == 'internal') {
+        onConfigChange(viewConfig);
+      }
+    },
+    // The function to specify which part of the store
+    // we want to subscribe to.
+    state => ({viewConfig: state.viewConfig, mostRecentConfigSource: state.mostRecentConfigSource}),
+  ), [viewConfigStoreApi, onConfigChange]);
+
+
+  useEffect(() => {
+    var authToken = 'mr-vitessce';
+    if (linkID !== null) {
+      const ws = new WebSocket('wss://irrmj4anbk.execute-api.us-east-1.amazonaws.com/production', ['Authorization', authToken, linkID]);
+      ws.addEventListener('open', (event) => {
+        console.log('Open', event);
+        setSocketOpen(true);
+      });
+      ws.addEventListener('connecting', (event) => {
+        console.log('Connecting', event);
+        setSocketOpen(false);
+      })
+      ws.addEventListener('error', (event) => {
+        console.log("Error", event)
+      });
+      ws.addEventListener('close', (event) => {
+        console.log("Close", event)
+      });
+      ws.addEventListener('message', (event) => {
+        console.log('Message', event);
+        if (event.data.includes(';')) {
+          if (event.data.split(';')[0] == id) {
+            console.log('Message from ourselves');
+          } else {
+            console.log('Message from server ');
+        //    console.log(JSON.parse(unescape(atob(event.data.split(';')[1]))))
+            if (sync) {
+              setConfig({
+                ...JSON.parse(unescape(atob(event.data.split(';')[1]))),
```

Comment:
probably skip the `unescape`, seems to have been deprecated

Reply:
Also, overall using `===` instead of `==` to make sure both type and value are matching.